tests/JustSaying.UnitTests/Messaging/Channels/SubscriberBusTests/WhenThereAreExceptionsInMessageProcessing.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon.SQS.Model;
+using JustSaying.AwsTools.MessageHandling;
+using JustSaying.TestingFramework;
+using NSubstitute;
+using Shouldly;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace JustSaying.UnitTests.Messaging.Channels.SubscriberBusTests
+{
+    public class WhenThereAreExceptionsInMessageProcessing : BaseSubscriptionBusTests
+    {
+        private ISqsQueue _queue;
+        private int _callCount;
+
+        public WhenThereAreExceptionsInMessageProcessing(ITestOutputHelper testOutputHelper)
+            : base(testOutputHelper)
+        {
+        }
+
+        protected override void Given()
+        {
+            _queue = CreateSuccessfulTestQueue("TestQueue", () =>
+            {
+                Interlocked.Increment(ref _callCount);
+                return new List<Message> { new TestMessage() };
+            });
+
+            Queues.Add(_queue);
+            Handler.Handle(null).ReturnsForAnyArgs(true);
+
+            SerializationRegister
+                .DeserializeMessage(Arg.Any<string>())
+                .Returns(x => throw new TestException("Test from WhenThereAreExceptionsInMessageProcessing"));
+        }
+
+        protected override async Task WhenAsync()
+        {
+            var cts = new CancellationTokenSource();
+            cts.CancelAfter(TimeoutPeriod);
+
+            var completion = SystemUnderTest.Run(cts.Token);
+
+            await Assert.ThrowsAnyAsync<OperationCanceledException>(() => completion);
+        }
+
+        [Fact]
+        public void TheListenerDoesNotDie()
+        {
+            _callCount.ShouldBeGreaterThanOrEqualTo(3);
```

Comment:
Where does the magic number `3` come from?

Reply:
Similarly this one is copied from [here](https://github.com/justeat/JustSaying/blob/efae8047ef7ecbf540957a55794f22a54c18f341/tests/JustSaying.UnitTests/AwsTools/MessageHandling/SqsNotificationListener/WhenThereAreExceptionsInMessageProcessing.cs#L89).
Not sure why 3, we have a similar test in ErrorHandlingTests that checks it keeps running if the Queue is throwing errors, I'll add a new one to check it keeps running if the message processing is erroring too.

Reply:
I've updated it to have 1 subscriber, and now it checks the _callCount is bigger than 1